client-shell/README.md

```diff
@@ -0,0 +1,108 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--->
+
+# Palisade Shell
+
+The Palisade Shell provides a CLI REPL for access to a cluster and its resources.
+
+> :information_source:
+In the case of the [Palisade examples](https://github.com/gchq/Palisade-examples/), the `context` is limited to just a `purpose` - one of `SALARY`, `DUTY_OF_CARE`, or `STAFF_REPORT`.
+Additionally, there are a number of pre-populated users (`Alice`, `Bob`, and `Eve`), resources (the pair of files `/data/local-data-store/employee_file{0,1}.avro`) and rules.
+All examples here are therefore written to be compliant with this example pre-populated data.
+
+## Sample Extract
+```shell script
+no-one@disconnected> connect pal://192.168.49.2:30094/?userid=Alice
+Connected to pal://192.168.49.2:30094/?userid=Alice
+
+Alice@192.168.49.2> register purpose=SALARY file:/data/local-data-store/
+6b6784e4-2383-40ac-92dc-7daa3b0a67b4
+
+Alice@192.168.49.2> cd 6b6784e4-2383-40ac-92dc-7daa3b0a67b4
+Selected 6b6784e4-2383-40ac-92dc-7daa3b0a67b4
+
+Alice@192.168.49.2#6b6784e4-2383-40ac-92dc-7daa3b0a67b4> ls
+file:/data/local-data-store/employee_file1.avro
+file:/data/local-data-store/employee_file0.avro
+
+Alice@192.168.49.2#6b6784e4-2383-40ac-92dc-7daa3b0a67b4> cat file:/data/local-data-store/employee_file0.avro
+Objavro.schema`{"type":"record","name":"Employee","namespace":"uk.gov.gchq.synt[+ 18086 characters]
+```
+
+
+## Commands
+Some effort has been made to make the Shell appear somewhat similar to a UNIX shell (bash, etc.).
+Subsequently, some commands have aliases to their UNIX counterparts.
+
+### Default Commands - `help`, `exit`
+Spring-shell-starter bundles some default commands, such as `help` and `exit`.
+Use the `help` command to list all available commands and their one-liner doc-string.
+Use the `exit` command to exit the shell.
+
+### Connect - `connect`
+Configure the backing [Java Client](../client-java/README.md) to point to a cluster address, as well as setting the `userId`.
+
+Suppose the target instance of Palisade is running locally in K8s and the ingress is exposed at `192.168.49.2:30094`, and we want to issue a request for data as "Alice", then our connect command would be:
+```shell script
+disconnected> connect pal://192.168.49.2:30094/?userid=Alice
+Connected to pal://192.168.49.2:30094/?userid=Alice
+```
+No checks are necessarily made that the server is available at this point, it is purely configuration for the client.
+This is described more in-depth in the [URL configuration](../client-java/README.md#URL) and [client properties](../client-java/README.md#Client%20properties) sections of the [client-java README](../client-java/README.md).
+
+### Register - `register`
+Given a `context` for accessing data, register a request for a `resourceId`.
```

Comment:
The short one liners are good for this, but as I have found in other places, I find I want/need a clear definition of these terms.  As a reader, I need a clear understanding of what is a context and what is a resource.  Could we consider a way to introduce this information into the documentation?

Reply:
It starts to become duplicated a lot and is better left the responsibility of the example. I think at the `clients` level, you'd assume some knowledge of palisade.

Reply:
good point